ccm_web/server/src/api/api.service.ts

```diff
@@ -120,6 +127,30 @@ export class APIService {
     return await sectionHandler.enrollUsers(sectionUsers)
   }
 
+  async enrollSectionExternalUsers (user: User, sectionId: number, sectionUsers: SectionExternalUserDto[]): Promise<string> {
+    // Create all requested users, noting failures
+    const adminRequestor = this.canvasService.createRequestorForAdmin('/api/v1/')
+    const adminHandler = new SectionApiHandler(adminRequestor, sectionId)
+    const canvasConfig = this.configService.get('canvas', { infer: true })
+    const createUserResponses = await adminHandler.createExternalUsers(sectionUsers, canvasConfig.newUserAccountID)
+    const newUsers = createUserResponses.filter(response => !isAPIErrorData(response)) as CanvasUser[]
+
+    // Invite only new users
+    const inviteService = new InvitationService(this.configService)
+    const inviteResults: string | null = await inviteService.sendInvitations(newUsers)
```

Comment:
We could lean in to the dependency injection/singleton approach supported by Nest instead of using `InvitationService` like an ordinary class. You'd have to create a `invitation.module.ts` and then import it and the service in `api.module.ts` (see the `CanvasService`/`CanvasModule` as an example).

Reply:
OK.  That's interesting.  I'll consider that.

Reply:
That worked out pretty well.  As I was using `InvitationService`, I had wanted to use it like `CanvasService` was used, but I wasn't certain about how to connect it.  I had that in the back of my mind as something I'd like to do, but it wasn't really urgent.  So, thank you for the suggestion and pointers.

🏁